frontend/controller/actions/identity.js

```diff
@@ -193,9 +193,12 @@ export default (sbp('sbp/selectors/register', {
       sbp('chelonia/contract/sync', contractIDs).then(async function () {
         // contract sync might've triggered an async call to /remove, so wait before proceeding
         await sbp('chelonia/contract/wait', contractIDs)
+        // similarly, since removeMember may have triggered saveOurLoginState asynchronously,
+        // we must re-sync our identity contract again to ensure we don't rejoin a group we
+        // were just kicked out of
+        await sbp('chelonia/contract/sync', identityContractID)
         await sbp('gi.actions/identity/updateLoginStateUponLogin')
-        // will only update it if it's different
-        await sbp('gi.actions/identity/saveOurLoginState')
+        await sbp('gi.actions/identity/saveOurLoginState') // will only update it if it's different
```

Comment:
Inline comments usually go on the previous line

Reply:
but it would break up that beautiful line of `await`s 😭